atlp-rwanda/e-commerce-mavericcks-bn
# [finsihes 187354259] added endpoint to get all products

## Purpose

The purpose of this pull request is to add new endpoints and functionality to the application:

- Added an endpoint to retrieve all products available in the database.
- Added an endpoint to retrieve a specific product by its ID.
- Added an endpoint for sellers to delete a product from their collection.

## Changes Made

- Added a new route and controller function for retrieving all products (`getAllProduct`).
- Added a new route and controller function for retrieving a product by ID (`getProductById`).
- Added a new route and controller function for sellers to delete a product (`deleteProduct`).
- Implemented functionality to exclude expired products when retrieving all products.
- Implemented error handling and response formatting for each endpoint.

## Testing Instructions

To test the changes:

1. Use an API testing tool like Postman or curl to send requests to the newly added endpoints.
2. For the "Get All Products" endpoint:
   - Ensure that all products are retrieved successfully.
   - Verify that expired products are excluded from the results.
3. For the "Get Product by ID" endpoint:
   - Send a request with a valid product ID and verify that the corresponding product is retrieved.
4. For the "Delete Product" endpoint:
   - Send a DELETE request with a valid product ID and seller authentication.
   - Verify that the product is successfully deleted from the seller's collection.


## Checklist
